### DISCUSSION TITLE
Play mp3 automatically and integration of HFS in Wordpress

### DISCUSSION THREAD

**Discussion by Officepoint** [rejetto/hfs#426] [General]
Hello!

I am new to using this program, but it is brilliant! I have searched for months reading and trying all kinds of programs like, Pydio, Nextcloud, Owncloud, OMV, Filebrowser, and others but nothing compares to the simplicity and ease of using this program. I hope with all my heart that support for it doesn't end quickly, at least if this is inevitable maybe there will be an early announcement of this. I have two questions.

 1 - Is it possible that when there is a folder with .mp3 files, there is also the possibility that the songs play automatically one after the other without pressing the mouse next each time or randomly? 

2 - How can HFS be integrated in the easiest way in a page from website that is created with Wordpress? Can you give an example?
  
Thank you very much, good health and all the best for 2024!

**Comment by rejetto** [comment]
hey, it's nice to know it's doing good for you!
HFS has a 20 years long career, so quite steady, in a way. Current embodiment only 2 years, tho. Since I made everything from scratch, I'm still focusing on its main job: file server. While being a media player is not its objective, functionality can be expanded with plugins. Some have made some players, and while most of them are probably basic, I'm not keeping an eye on them, so maybe you want to check.
What you are asking is not a core feature, but I'll think about it.
Please, specify if you are currently trying to play the mp3 files clicking "open" or "show".
I just noticed "show" has misaligned arrows. 😰

2. I guess that HFS can be integrated in any website using in iframe, or putting a link to it. Is that it?

**Reply by Harry-nl** [reply]
Hi I also completed new here, and set it up as normal private server, but from my social network site I use a link to the the server. you need a dedicated Ip for it to get that's really working. For my members I have a secret login name and code.  And this way it working fine.

**Comment by Officepoint** [comment]
Hi Rejetto,

It is normal that the focus is primarily on to the functionality as a
server, but I was wondering if it is not too complicated, if this
function can be implemented, or at least some information if someone
has found a solution to this problem. I found out about HFS a few
years ago but somehow at that time I had the impression that it was an
abandoned project, but it's also true that I didn't really look into
it. Now I'm really glad I discovered it and I'm sure it's exactly what
I need, although after hours and hours of trying I still haven't
figured out how to add permissions separately on folders so that some
users have access to them and other users It is natural that the focus
is primarily on the functionality as a server, but I was wondering if
it is not too complicated, if this function can be implemented, or at
least some information if someone has found a solution to this
problem. I found out about HFS a few years ago but somehow at that
time I had the impression that it was an abandoned project, but it's
also true that I didn't really look into it. Now I'm really glad I
discovered it and I'm sure it's exactly what I need, although after
hours and hours of trying I still haven't figured out how to add
permissions separately on folders so that some users have access to
them and other users don't.
It's something similar to this post:

https://github.com/rejetto/hfs/discussions/216

I've tried for hours to set permissions per folder but have failed,
even though I've read almost all 6 pages of questions.
Honestly I still don't understand the order in which they apply, I
mean Who can download, Who can see, etc. What I have noticed is that
if they are set wrong they override each other and if you don't
remember how they were set, I don't know how you can go back to the
original settings. Maybe an option like, Revert to orginal settings...
would have been helpful. For the moment I'm only interested in these
options, to set per user which folder is seen and also to download it.
I'm thinking that maybe 2 or 3 practical examples would be useful, or
a video, even using Masks, which as you say is not that simple. But
who knows, maybe it's easier to learn a few syntaxes and then clone
them each time changing names of users and folders.

I'll try again.
Thank you very much for your answer!
Al the best, Mihai!

**Comment by rejetto** [comment]
you'll be glad to know that I already did the auto-play feature. It will arrive in 0.52 .
I will start publishing 0.52 as beta as soon as 0.51 turns stable.

about permissions, if you need a folder to be accessed only by some accounts, you just click on the "who can download" and then "select accounts". That's it. Does this cover your needs? Don't change other permissions unless you need more than what you are getting. I don't think you need masks.

About resetting to default, when you click on permissions (like "who can download"), one of the option will start with "default:" or "as parent:", that is the default, if you need to restore it.
But if you really touched a lot of stuff, you can even delete the folder and re-add it.

**Comment by Officepoint** [comment]
Hi Rejetto!

This is extremely good news what I hear.
Thank you very much, and I'm very happy for the implementation!
I can't wait to test it out.
I am now using HFS 0.51 beta 7.  I wanted to see how the latest version
works although I know the beta version, is not the final one. What do you
think, would it be better to use 0.51 stable until 0.52 is ready?
Another suggestion would be if you could implement the same function for
pictures, nothing fancy, just a play button. Something more complicated
would be if there were some transition effects for pictures like fade,
wipe, push, swipe etc. More transition effects can be found in PowerPoint.
If they were not extremely difficult to implement I would implement them
all.
It would be very nice to press the play button and watch the pictures with
various transition effects while a song plays in the background.
It's just an idea, as you said the main focus is not this, but maybe who
knows, at some point you will discover an easy way to implement that
doesn't cost much time.

As for the permissions on the folders, Who can download..., etc, I'll try
again, just with this setting without having to go into the others, Who can
see. etc...
I've figured out that the only way you can go back to the "original
settings" is to delete the folder and reopen it.

All the best, and thanks again!
Mihai!
Keep in touch!

**Comment by rejetto** [comment]
if should you use a beta or not, it depends on your willingness to face potentially more problems, while getting more features.
I say "more" because no software is 100% safe, so betas are just more dangerous.
And of course HFS3 needs still time before being called mature.
Using beta versions is a huge help for the project, I'm going to put this in the "contribute" section.

**Comment by rejetto** [comment]
but it's easy for me to provide you a preview of 0.52 if you want to try that feature

**Comment by Officepoint** [comment]
Hi,

Sure, I can try it to see if it works.

Thank you!

**Comment by Officepoint** [comment]
Hi Rejetto!

I'm still waiting if you want to test the player.
I'm already downloaded the version 0.51.0-beta10, but i did not find the functions for the player.
And another question, which I asked a little above, is there any chance that this playback system will be implemented also for pictures?
The playback system doesn't have to be super complicated, and if it can't include some effects at least there should be a simple play button. 

Thanks!

**Comment by rejetto** [comment]
Hey, I forgot about this. Been making fixes. Will try to provide asap a preview of 0.52
Yes, it supports pictures. Of course it's basic. 
Keep you posted.

**Comment by Officepoint** [comment]
Ok, no problem, then i will try when 0.52 will arrive.
I'm very glad to hear that the player will be able to play pictures automatically, even without effects, although it would have been extremely nice to have at least 4-5 effects.
Do you have any idea / estimation in time when 0.52 can be tested?

Thanks!

**Comment by rejetto** [comment]
maybe within a week

**Comment by rejetto** [comment]
here it is!
https://github.com/rejetto/hfs/releases/tag/v0.52.0-alpha1

**Comment by Officepoint** [comment]
Thank you Rejjeto!

It's working very nice! I like it very much!
I have two suggestions.

It is possible that the mp3 player will have both of these functions:

1 - A button - random - that when it is active randomly selects songs
2 - A button - repeat - which when it is active will play the songs or only one of them continuously

And for the playback of pictures it is possible to play them through a fade in - fade out effect, when i choose this effect. It would be great if there will be another effect or two, but if more than one is not possible, at least one - fade in - fade out, except the basic one.

All the best!

**Comment by rejetto** [comment]
i will try to have random and repeat.
effects i'm not sure.
look at this https://github.com/rejetto/hfs/releases/tag/v0.52.0-alpha2

**Comment by Officepoint** [comment]
Hi!
I'm glad to hear this good news! 
If you could also implement 2-3 effects for pictures it would be perfect, at least from my point of view. Anything else would be unnecessary.
Unfortunately I don't see the buttons anywhere. The autoplay function is functional, but I don't see any option to select randomly or continuously.
A small suggestion, it would be helpful to have a small pause after finishing each song and starting the next. I think 3 seconds is enough.
Thanks!

**Comment by Officepoint** [comment]
Hi!
Any suggestion to see the buttons on mp3 player?
Thanks!

**Comment by rejetto** [comment]
sorry, i forgot to answer!
i didn't say you have the buttons now.
> i will try to have random and repeat

"will" is for future, and "try" is not guaranteed 😄

>  it would be helpful to have a small pause after finishing each song

that should be optional, and i'm not for having that many options at this moment

**Comment by Officepoint** [comment]
Ok, no problem!
My bad!
I will wait.
All the best!

**Comment by rejetto** [comment]
would you enjoy a preview?
[hfs-windows.zip](https://github.com/rejetto/hfs/files/14152601/hfs-windows.zip)

<img width="278" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/02deb038-7d1c-48e4-a054-3350a67d588c">
<img width="277" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/e99a8de4-55b7-4b9c-b86f-0121248cdd44">

**Comment by Officepoint** [comment]
Hi!
Yes of course!
I tested it and it works very nicely!
A small suggestion, if it's not too difficult to implement. If it's
difficult no problem, it can stay that way, I'm so glad they are functional.
When I activate one or both of the functions a sign like V might appear on
the right, instead of the buttons disappearing.
I have attached a small example in the image below.

Thank you very much Rejjeto!

![Activation - shuffle - repeat](https://github.com/rejetto/hfs/assets/94564442/adbb6b32-5ad0-4648-ba48-548fef57450d)

**Comment by rejetto** [comment]
email attachments don't work on github

**Comment by Officepoint** [comment]
I added the image in the previous post.

Thank you!

**Comment by rejetto** [comment]
i guess the current visualization is not clear enough....?
here shuffle is enabled
<img width="254" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/6cf3b478-9030-4da6-b33b-2f323efb1a8d">

**Comment by Officepoint** [comment]
I realized now, it's because of the theme, on dark you can see it very well, but on white you can't see it anymore, that's why I was saying that when I activate a button it immediately disappears, the same happens with both when they are activated.
From my point of view it's easier to have a V appear on the right when it's activated and disappear when it's not activated, but as I said, it's not a big problem at all if it stays as it is now.
One question, how to put the picture in the background?

https://github.com/rejetto/hfs/assets/94564442/853033bf-5028-445b-b1cc-6b598cc607f4

**Comment by rejetto** [comment]
well, that's a bug! keep you posted

The picture is automatically displayed if you have a jpg in the folder named cover, or albumart, or folder. These are the names that i know to be standard.

**Comment by Officepoint** [comment]
Yes, it works superbly with the cover too!
If I put the cover this image will load by default for all users?
![Cover for mp3 player](https://github.com/rejetto/hfs/assets/94564442/29456b86-041b-424f-a364-a371333486cd)

**Comment by rejetto** [comment]
it is used for all users that see that file in the list

**Comment by rejetto** [comment]
this version is fixing the problem with white theme
https://github.com/rejetto/hfs/files/14157522/hfs-windows.zip
can you confirm?

**Comment by Officepoint** [comment]
Hi!
It's working very good!
Thanks!!
![Screenshot_1](https://github.com/rejetto/hfs/assets/94564442/6b20c0c7-3851-44c0-852e-609b92cf11da)

**Comment by Officepoint** [comment]
Hi!
There is a way to select all files by one click, and not one by one?
I want to create a playlist with all songs, and for that i create a virtual folder and i will put inside all songs.
There is no problem with 10 songs, but with 300 it will be a lot of clicks.

![Select all files](https://github.com/rejetto/hfs/assets/94564442/8a3dd850-a43f-4f2b-8b72-1481eb8982ed)

Thank you!

**Comment by rejetto** [comment]
please, for different topics create a new discussion, or it becomes difficult to handle.

you are doing it wrong: if you want to have all files, bind the virtual folder to the one on disk, using the "load content from disk", and all files will be displayed

**Comment by Officepoint** [comment]
Hi!
I posted on this topic because it was related to mp3 player. But it's no problem, I'll open another topic. 

Thanks!

**Comment by majkinetor** [comment]
HI @rejetto 

I am on 0.52. alfa2 and this is nowhere to be found in it. I tried white theme fix but in it shared files no longer work for some reason (shows Not found). Am I missing something ?


I can not see File show dialog anywhere, playing single mp3 just opens standard html player and once the song is over it stops.

**Reply by majkinetor** [reply]
Ah, nvm, I figure out, I need to click Show in file menu. Clicking the file itself doesn't do that.

One problem I found is it doesn't work with FLAC format. HTML player plays it but in file menu there is no Show but only Download option. I would appreciate if you could support it.

**Comment by rejetto** [comment]
Oh, right, i didn't know flac was supported.
will add to next release.
thanks
